src/pages/index.js

```diff
@@ -166,44 +227,99 @@ addButton.addEventListener("click", () => openModal(addModal));
 
 addCloseButton.addEventListener("click", () => closeModal(addModal));
 
-//handles edit profile submit
+//handles edit profile submit    ---- FIX RELOAD NAME
 function handleEditProfileSubmit(evt) {
   evt.preventDefault();
-  profileNameEl.textContent = editProfileNameInput.value;
-  profileDescriptionEl.textContent = editProfileDescriptionInput.value;
-  closeModal(editModal);
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Saving...");
+
+  api
+    .editUserInfo({
+      name: editProfileNameInput.value,
+      about: editProfileDescriptionInput.value,
+    })
+    .then((user) => {
+      profileNameEl.textContent = user.name;
+      profileDescriptionEl.textContent = user.about;
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+      closeModal(editModal);
```

Comment:
`      closeModal(editModal);
`

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**